src/basic/selinux-util.c

```diff
@@ -59,14 +70,15 @@ bool mac_selinux_enforcing(void) {
 #if HAVE_SELINUX
         if (_unlikely_(cached_enforcing < 0)) {
                 cached_enforcing = security_getenforce();
-                if (cached_enforcing == -1)
-                        log_error_errno(errno, "Failed to get SELinux enforced status, continue in enforcing mode: %m");
-                else
-                        log_debug("SELinux enforcing state cached to: %s", cached_enforcing ? "enforcing" : "permissive");
+                if (cached_enforcing < 0) {
+                        log_debug_errno(errno, "Failed to get SELinux enforced status, continuing in enforcing mode: %m");
```

Comment:
@poettering why change to `log_debug_errno()` from `log_error_errno()` (did you meant `log_enforcing_errno()` ?) ? I think a failure of `security_getenforce()` should be logged.

Reply:
Generally we have the rule that "library-like" calls should not log above LOG_DEBUG level, and leave that to their callers. Otherwise we'll get duplicate logging. This stuff is clearly library-like and hence should not log itself. It was a mistake that we had this here...

Or in other words: the code that calls this function should be the one that logs about this beyond LOG_DEBUG if it's something to warn about, or alternatively propagate it further so that eventually something further up the stack logs...

Also see:

https://systemd.io/CODING_STYLE/#error-handling

Reply:
But callers of `mac_selinux_enforcing()` cannot distinguish between a successful "enforce"/true read and a "enforce"/true fallback (when `security_getenforce()` fails)

Reply:
hmm, true, maybe this should be rework to return `int` then, i.e. < 0 on error, == 0 on "no", > 0 on "yes"